crdt/sequence.py

```diff
@@ -0,0 +1,247 @@
+from functools import cmp_to_key
+from re import S
+import uuid
+from enum import Enum
+
+from crdt.gset import GSet
+from crdt.gcounter import GCounter
+
+class Sequence():
+    """
+    Sequence is a CRDT that represents an ordered set of objects and supports insertion
+    and removal at arbitrary positions in the set.
+    """
+
+    def __init__(self, id=uuid.uuid4()):
+        self.id = id
+        self.operations = GSet()
+        self.clock = GCounter(self.id)
+        self.sequence = []
+
+    def compare_operations(self, a, b):
+        """
+        Compares this Operation to another Operation to determine ordering.
+        """
+        return a.owner.compare(b.owner)
+
+    def index_transform(self, position):
+        """
+        Returns the transformed index, which is the actual index of the object in the
+        sequence, taking into account tombstones.
+        """
+        if position < 0 or position >= len(self.sequence):
+            raise IndexError("Index out of bounds")
+
+        index = -1
+        while position >= 0:
+            index += 1
+            if not self.sequence[index].tombstone:
+                position -= 1
+        return index
```

Comment:
This is required so that the caller doesn't have to know anything about operation IDs when performing inserts and removals and they can just pass in a normal index.